src/dynamic_batch_scheduler.cc

```diff
@@ -633,31 +633,41 @@ DynamicBatchScheduler::DelegateResponse(
   std::lock_guard<std::mutex> lock(completion_queue_mtx_);
   completion_queue_.emplace_back();
   auto queue_slot = &completion_queue_.back();
-  const auto key = request->CacheKey();
+  // Cache plumbing
+  auto key = request->CacheKey();
+  uint64_t lookup_ns =
+      request->CacheLookupEndNs() - request->CacheLookupStartNs();
+  if (!request->CacheKeyIsSet()) {
+    key = "";
+    LOG_ERROR << "Request cache key was not set correctly.";
```

Comment:
Does this error gets printed when the model is not using any caching too?

Reply:
Good catch. I think you're right, but it's actually surprisingly hard to hit this code path with caching disabled, so I didn't see it. 

Just as fun fact, it looks like you would need to have:
- dynamic batching disabled (which means you need `--disable-auto-complete-config` because you can't specify default scheduler in config file and autocomplete chooses dynamic_batching) 
- `preserve_ordering` to be true, but it is a [property of dynamic_batching](https://github.com/triton-inference-server/common/blob/5313d0be99204c89ab696023534043f2ba535868/protobuf/model_config.proto#L1104-L1135) ? So I don't understand how this path would get hit. This is true even in 21.09 before cache was added: https://github.com/triton-inference-server/server/blob/39b93374b88ecba2c4912323ed6406c13d0bc6b7/src/core/dynamic_batch_scheduler.cc#L153-L156

Reply:
Moved error checking logic into cache_enabled section. These checks indicate a bug if they are hit, so it shouldn't be spamming users unless something is wrong.